### PR TITLE
Add audit logs for when a group is created

### PR DESCRIPTION
**Pull request by Ellen-Yi-Dong**
## Link(s) to Jira
- Add Audit Logs for when group is created
- Added Unit test for group 

## Description of Intent of Change(s)
The what, why and how.

## Local Testing
How can the feature be exercised?
How can the bug be exploited and fix confirmed?
Is any special local setup required?

## Checklist
- [ ] if API spec changes are required, is the spec updated?
- [ ] are there any pre/post merge actions required? if so, document here.
- [ ] are theses changes covered by unit tests?
- [ ] if warranted, are documentation changes accounted for?
- [ ] does this require migration changes?
  - [ ] if yes, are they backwards compatible?
- [ ] is there known, direct impact to dependent teams/components?
  - [ ] if yes, how will this be handled?

## Secure Coding Practices Checklist Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Practices Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

File: rbac/management/group/view.py [hunk 1]
```diff
@@ -250,7 +251,12 @@ def create(self, request, *args, **kwargs):
             }
         """
         validate_group_name(request.data.get("name"))
-        return super().create(request=request, args=args, kwargs=kwargs)
+        create_group = super().create(request=request, args=args, kwargs=kwargs)
+
+        if status.is_success(create_group.status_code):
+            auditlog = AuditLog()
+            auditlog.log_create(request, AuditLog.GROUP)
+            return create_group
```
Comment (astrozzc) [inline rbac/management/group/view.py:259]: I think this should be moved out of the If statement

Comment (lpichler) [reply rbac/management/group/view.py:259]: it looks this needs to be fixed also for roles: https://github.com/RedHatInsights/insights-rbac/blob/master/rbac/management/role/view.py#L218

Comment (lpichler) [reply rbac/management/group/view.py:259]: However, it might not be necessary since it only fails in cases where it doesn't return the entire object.

**Review by astrozzc — APPROVED** [review]
LGTM

**Review by lpichler — APPROVED** [review]

**Review by petracihalova — COMMENTED** [review]
I know it is not part of this PR but it is some reason why we use for groups the protected attribute `_data`?
https://github.com/RedHatInsights/insights-rbac/blob/409d26f8d197394dbef50d69153bc466132b0b83/rbac/management/audit_log/model.py#L84-L86

**Comment by Ellen-Yi-Dong** [comment]
Oh! The protected data was used because before groups didn't have the .data aspect (or one that I saw before), but I was testing out my code for the follow up to this PR, and I saw the .data is now being recognized, so I'll update that! Thank you so much!

File: rbac/management/audit_log/model.py [hunk 2]
```diff
@@ -64,6 +64,7 @@ class AuditLog(TenantAwareModel):
 
     def get_tenant_id(self, request):
         """Retrieve tenant id from request."""
+        print("entering tenant")
```
Comment (lpichler) [inline rbac/management/audit_log/model.py:67]: this is temporal ?

Comment (Ellen-Yi-Dong) [reply rbac/management/audit_log/model.py:67]: Hello Libor! 
Yes! There's something from the IQE test that I want to test, but i no longer have access to the ephemeral env, so Im hoping that these print statements can help when vinicius runs the env!

File: rbac/management/audit_log/model.py [hunk 3]
```diff
@@ -96,15 +98,16 @@ def get_resource_item(self, r_type, request, *args, **kwargs):
             return None
 
         elif r_type == "principal":
+            print("entering principal")
             principal_object = get_object_or_404(Principal, username=request.user.username, tenant=verify_tenant)
```
Comment (vbelchio) [inline rbac/management/audit_log/model.py:102]: After the tests in a fresh database, @Ellen-Yi-Dong and I found the reason PR_Checks is not passing. It is related to getting the principals from the database once the user requester is not a principal yet.
The user requester just will be inside the database as a principal just after being inside a group, so when we are creating a group using a user requester that is not a principal yet, we are getting 404.

Comment (Ellen-Yi-Dong) [reply rbac/management/audit_log/model.py:102]: Attempting to fix this issue from this ticket: https://issues.redhat.com/browse/RHCLOUD-33727

Will create separate PR that will most likely be merged first

**Comment by Ellen-Yi-Dong** [comment]
PR to remove principal check here: https://github.com/RedHatInsights/insights-rbac/pull/1137

**Review by vbelchio — APPROVED** [review]

**Review by lpichler — APPROVED** [review]

**Review by petracihalova — APPROVED** [review]

File: rbac/management/audit_log/model.py [hunk 4]
```diff
@@ -82,11 +81,11 @@ def get_resource_item(self, r_type, request, *args, **kwargs):
             return role_object_id, role_object_name
 
         elif r_type == AuditLog.GROUP:
-            if request._data is not None:
+            if request.data != {}:
                 group_object = get_object_or_404(Group, name=request.data["name"], tenant=verify_tenant)
             else:
                 group_uuid = kwargs["kwargs"]["uuid"]
-                group_object = get_object_or_404(Group, uuid=group_uuid)
+                group_object = get_object_or_404(Group, uuid=group_uuid, tenant=verify_tenant)
```
Comment (astrozzc) [inline rbac/management/audit_log/model.py:88]: Nit: here tenant is not necessary because uuid is unique for each group

**Pull request merged**